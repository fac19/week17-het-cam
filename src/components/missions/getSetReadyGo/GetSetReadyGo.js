import React from "react";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import { makeStyles } from "@material-ui/core/styles";
import styled from "styled-components";
import { accentColour, fontColour } from "../../global/colours";

const useStyles = makeStyles((theme) => ({
  root: {
    position: "relative",
    marginLeft: "0.4rem",
    marginRight: "0.4rem",
  },
  inactive: {
    color: fontColour,
  },
  active: {
    color: accentColour,
  },
}));

const StyledWrapper = styled.div`
  display: flex;
  flex-direction: row;
  font-family: "Josefin Sans", sans-serif;
  text-align: center;
  align-items: center;
  justify-content: center;
  margin-top: 0;
`;

const StyledText = styled.p`
  font-size: 1.5rem;

  @media (min-width: 600px) {
    font-size: 2rem;
  }
`;

const GetSetReadyGo = (props) => {
  const classes = useStyles();
  return (
    <StyledWrapper>
      <StyledText className={classes.active}>Get Set</StyledText>
      <ArrowForwardIosIcon
        className={
          props.missionState === "ready" || props.missionState === "go"
            ? classes.active
            : classes.inactive
        }
      />
      <StyledText
        className={
          props.missionState === "ready" || props.missionState === "go"
            ? classes.active
            : classes.inactive
        }>
        Ready
      </StyledText>
      <ArrowForwardIosIcon
        className={
          props.missionState === "go" ? classes.active : classes.inactive
        }
      />
      <StyledText
        className={
          props.missionState === "go" ? classes.active : classes.inactive
        }>
        Go
      </StyledText>
    </StyledWrapper>
  );
};

export default GetSetReadyGo;
