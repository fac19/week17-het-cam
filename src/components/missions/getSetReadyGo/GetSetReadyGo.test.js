import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import GetSetReadyGo from "./GetSetReadyGo";

test("GetSetReadyGo component is styled", () => {
  const tree = renderer.create(<GetSetReadyGo />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
});

test("GetSet is rendered", () => {
  render(<GetSetReadyGo missionState={"get"} />);
  screen.getByText("Get Set");
});
