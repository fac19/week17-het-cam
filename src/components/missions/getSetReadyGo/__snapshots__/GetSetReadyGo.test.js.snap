// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetSetReadyGo component is styled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-family: "Josefin Sans",sans-serif;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 0;
}

.c1 {
  font-size: 1.5rem;
}

@media (min-width:600px) {
  .c1 {
    font-size: 2rem;
  }
}

<div
  className="c0"
>
  <p
    className="c1 makeStyles-active-3"
  >
    Get Set
  </p>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root makeStyles-inactive-2"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
    />
  </svg>
  <p
    className="c1 makeStyles-inactive-2"
  >
    Ready
  </p>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root makeStyles-inactive-2"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
    />
  </svg>
  <p
    className="c1 makeStyles-inactive-2"
  >
    Go
  </p>
</div>
`;
